next/components/footer/footer.tsx

```diff
@@ -15,7 +15,9 @@ interface FooterProps {
 export function Footer({ menu }: FooterProps) {
   // Only show the menu items that match the current locale:
   const { locale } = useRouter();
-  const filteredItems = menu.items?.filter((link) => link.langcode == locale);
+  const filteredItems = menu.items?.filter(
+    (link) => link.langcode?.id == locale,
+  );
```

Comment:
The new version of graphql_compose introduces the `langcode` as a properly shaped `language` object, so we have to use langcode.id here